src/pages/CreateEventPage/CreateEventPage.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react';
+import { CreateEventWrapper, Form } from './styled';
+import { formatISO, format } from 'date-fns';
+import { createEvent } from '../../api/events';
+import TextField from '../../components/TextField';
+import DateField from '../../components/DateField';
+import TimeField from '../../components/TimeField';
+import SelectField from '../../components/SelectField';
+import TextareaField from '../../components/TextareaField';
+import Alert from '../../components/Alert';
+
+function CreateEvent() {
+  const [name, setName] = useState('');
+  const [image, setImage] = useState('');
+  const [imageAlt, setImageAlt] = useState('');
+  const [date, setDate] = useState(format(Date.now(), 'yyyy-MM-dd'));
+  const [starttime, setStarttime] = useState(format(Date.now(), 'HH:mm'));
+  const [endtime, setEndtime] = useState(format(Date.now(), 'HH:mm'));
+  const [description, setDescription] = useState('');
+  const [eventType, setEventType] = useState('Choose event type');
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [createWasSuccess, setCreateWasSuccess] = useState(false);
+  const [createFailed, setCreateFailed] = useState(false);
+  const eventOptions: string[] = [
+    'Subject matter event',
+    'Coding event',
+    'Project presentation',
+    'Workshop',
+  ];
+
+  function submitEvent() {
+    setCreateFailed(false);
+    setCreateWasSuccess(false);
+    createEvent(
+      formatISO(Date.parse(date + ' ' + starttime)),
+      formatISO(Date.parse(date + ' ' + endtime)),
+      name,
+      description,
+      firstName,
+      lastName,
+      email
+    )
+      .then((result) => {
+        if (result) {
+          setCreateWasSuccess(true);
+        } else {
+          setCreateFailed(true);
+        }
+      })
+      .catch(() => {
+        setCreateFailed(true);
+      });
```

Comment:
There's some chances for a race-condition here, if the user clicks the submit button multiple times. You could use the `useAsyncData` hook, but we should probably create a more general `usePromise` hook that can be used without necessarily expecting any data back.

Reply:
Solved it differently, please help if you think useAsync is better.

Reply:
Your solution is good enough 👍.

Reply:
If I read this correctly `setClicked(true)` will run as soon as the button is clicked, then inside this fuction the `createEvent` promise is dispatched (but not awaited) and `setClicked(false)` is immediately called. For this to properly protect against the race-condition you need to either have `setClicked(false)` inside a final `then` and `catch` handler on the promise or use async await to make it procedural (inside a "try catch finally"-block).

Reply:
I don't see any `setClicked` calls?

Reply:
Use ctrl+f and you'll see, line 57 and 164

Reply:
The mechanism for avoiding race condition is disabling the submit button when the user clicks

Reply:
I added .finally, should have seen this.